### PR TITLE
mongoDB as storage

### PR DESCRIPTION
**Pull request by Pedrisimo**

### CODE REVIEW

File: controllers/items.js [hunk 1]
```diff
@@ -1,85 +1,58 @@
...
 exports.getAction = function (request, response) {
-    fs.exists(config.database.path, function (isExists) {
-        response.writeHead(200, {'Content-Type': 'application/json'});
-
-        if (false === isExists) {
-            response.end(JSON.stringify([]));
-
-        } else {
-            fs.createReadStream(config.database.path).pipe(response);
+    let docs;
```
Comment (mikalai-sauchanka) [inline controllers/items.js]: move the variable inside the getDBRecs' callback body

File: controllers/items.js [hunk 2]
```diff
@@ -1,85 +1,58 @@
...
+    getDBRecs((error, data) => {
+        if (error) {
+            docs = JSON.stringify(error.name + " : " + error.message);
         }
+        else {
+            docs = JSON.stringify(data);
+        }
+        console.log("DB Response: " + docs);
+        response.writeHead(200, {'Content-Type': 'application/json'});
```
Comment (mikalai-sauchanka) [inline controllers/items.js]: Below you used 503 http status code, which is in an error response, and this is correct. Use same approach here

File: controllers/items.js (hunk 1 above)
Comment (mikalai-sauchanka) [inline controllers/items.js]: Same in the methods below

File: controllers/items.js [hunk 3]
```diff
@@ -1,85 +1,58 @@
...
-                response.end(JSON.stringify({error: 'Can\'t save data. Please see server\'s console output for details.'}));
-
-            } else {
-                response.writeHead(200, {'Content-Type': 'application/json'});
-                response.end(JSON.stringify(postData));
-            }
-        });
+    setDBRecs(postData, function(error, data) {
+        if (error) {
+            docs = JSON.stringify(error.name + " : " + error.message);
```
Comment (mikalai-sauchanka) [inline controllers/items.js]: JSON.stringify is unnecessary here

File: controllers/items.js [hunk 4]
```diff
@@ -1,85 +1,60 @@
...
+        let responseData;
+        if (error) {
+            responseData = error.name + " : " + error.message;
+            response.writeHead(503, {'Content-Type': 'application/json'});
+            response.end(JSON.stringify({responseData: 'Can\'t get data. Please see server\'s console output for details.'}));
         }
+        else {
+            responseData = JSON.stringify(data);
+        }
+        console.log("DB Response: " + responseData);
```
Comment (mikalai-sauchanka) [inline controllers/items.js]: You need to put the lines below into the else condition, otherwise it will be executed in case of error

**Comment by Pedrisimo** [comment]
Mentioned issues fixed in https://github.com/Pedrisimo/epam-tasks-app2/pull/7/commits/fbeb6cca22322dd80751dc0716597e18a6e007a4

**Pull request merged**